src/main/java/org/interledger/Condition.java

```diff
@@ -0,0 +1,66 @@
+package org.interledger;
+
+import java.util.Arrays;
+
+/**
+ * The execution condition attached to all transfers in an Interledger payment.
```

Comment:
Might be worth adding a link to the [crypto-conditions spec](https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/), and even possibly copying some of the Preimage-sha-256 documentation into here so that it's more clear what a "Condition" is?